### ISSUE TITLE
Create validate typescript library node package

### ISSUE THREAD

**Issue by mjtravers** [fecgov/fecfile-validate#34]
Create a typescript library to perform in-line validation using the JSON schema. The library should be imported using npm package manager pulling directly from GitHub.

- Resource [how-to](https://itnext.io/step-by-step-building-and-publishing-an-npm-typescript-package-44fe7164964c) on setting up a Typescript library for npm
- Use [Ajv](https://ajv.js.org/) library to perform JSON schema validation

API Function

function validate(schema, data)

Returns an array of errors or an empty array if none are found.

Validation error information for a single schema property:
```
 @typedef {object} ValidationError
 @property {string} path - property name that failed the validation test
 @property {string} keyword - type of validation check
 @property {Record<string, any>} params - additional info per the type of validation check
 @property {string | null} message - human readable message describing validation error
 ```

Error example:
```
[{
    keyword: "required",
    message: "must have required property 'form_type'",
    params: {
        missingProperty: "form_type",
    },
    path: "",
}]
```

**Acceptance Criteria**
- npm package formatted with validate() functions as described above
- package can be installed in fecfile-web-app from fecfile-validate GitHub using npm package manager
- unit tests
- JSDoc comments

**Comment by AureliaKhorsand** [comment]
**Refinement Meeting 2/9/22:** Task intended for Robert. Matt will expand on QA portion of this ticket when it comes through code review. Pointed at 13 for padding.

**Comment by mjtravers** [comment]
There were HTTP size bundle issues with the way the original ticket acceptance criteria was written. I updated the Acceptance Criteria to match the API changes necessary to better manage the large number of JSON schema documents that will be pulled into the front-end in a more efficient manner with code tree shaking and Angular lazy loading strategies.

**Comment by mjtravers** [comment]
The SonarCloud Code Analysis is reporting 0% code coverage for the new javascript code. This is inaccurate as there are unit tests for this code. The issue is that CircleCI is not configured to send code coverage reports to SonarCloud for the javascript code as it is for the python code. This ticket has been generated to fix this issue: #60

**Comment by mjtravers** [comment]
For QA, since CircleCI has not been set up to run unit tests for javascript, the tests were run locally and the results posted via screenshot. [Here is the actual test code](https://github.com/fecgov/fecfile-validate/blob/develop/fecfile_validate_js/tests/index.test.ts)
![node-validate-test.png](https://images.zenhubusercontent.com/61b9fc72d99b6f786e0d798c/0f313d1a-ad4d-4d68-bd7b-2ab214e9ab3a)
